### PR TITLE
Implement static iteration matrix for a single beam element

### LINKED ISSUE #73: Implement tangent stiffness for iteration matrix for single static beam

### PR DESCRIPTION
**Pull request by faisal-bhuiyan**
Resolves #73.

Namely the iteration matrix (under static loading) for a beam element was implemented based on the development provided in the accompanying [theory guide](https://github.com/michaelasprague/OpenTurbineTheory/tree/main/BeamTheoryNotes) and unit tests provided in the [mathematica implementation](https://github.com/michaelasprague/OpenTurbineTheory/blob/main/mathematica/StaticIterationMatrix.nb). 

As a follow up work, we need to adopt this to work with our field and mesh data structures.

### CODE REVIEW

File: src/gebt_poc/solver.cpp
Comment (ddement) [inline src/gebt_poc/solver.cpp:323]: Perhaps each of the O, P, and Q matrices should be assembled in their own function?  This would help with independent testing and make things a little more readable than a single big function.

File: tests/unit_tests/gebt_poc/test_solver.cpp
Comment (ddement) [inline tests/unit_tests/gebt_poc/test_solver.cpp:393]: I think we might want some "null" tests like we have for the residual matrix.  This will make refactoring a bit easier.

**Review by ddement — APPROVED** [review]
Looks good.  Is there some way to move some of our expected results into a header file?  These matrices are getting pretty big to the point that they make the tests less readable.  Even something as simple as making them global variables under a "golds" namespace or something might be enough for now.

File: tests/unit_tests/gebt_poc/test_solver.cpp
Comment (alanw0) [inline tests/unit_tests/gebt_poc/test_solver.cpp:487]: I'm having trouble knowing when there should be 6 values per row (per node) vs when there should be 7 values per row. But that's probably not a significant issue, probably just me not keeping up with the math development lately.

Comment (faisal-bhuiyan) [reply tests/unit_tests/gebt_poc/test_solver.cpp:487]: @alanw0 Yup, this is a direct consequence of the SO3-based mathematical development we are using. Please take a look at the following excerpt - basically position and displacements would contain 7 components/node, while velocity, acceleration will contain 6 components/node.

image

Comment (faisal-bhuiyan) [reply tests/unit_tests/gebt_poc/test_solver.cpp:487]: Please feel free to ask more question - we should get one approval before we merge this PR

**Review by alanw0 — APPROVED** [review]

**Pull request merged**
